src/clib/pio_file.c

```diff
@@ -481,103 +595,7 @@ int PIOc_sync(int ncid)
 
     LOG((1, "PIOc_sync ncid = %d", ncid));
 
-    /* Get the file info from the ncid. */
-    if ((ierr = pio_get_file(ncid, &file)))
-        return pio_err(NULL, NULL, ierr, __FILE__, __LINE__);
-    ios = file->iosystem;
-
-#ifdef _ADIOS
-    if (file->iotype != PIO_IOTYPE_ADIOS)
-    {
-#endif
-
-    /* Flush data buffers on computational tasks. */
-    if (!ios->async || !ios->ioproc)
-    {
-        if (file->mode & PIO_WRITE)
-        {
-            wmulti_buffer *wmb, *twmb;
-
-            LOG((3, "PIOc_sync checking buffers"));
-            wmb = &file->buffer;
-            while (wmb)
-            {
-                /* If there are any data arrays waiting in the
-                 * multibuffer, flush it to IO tasks. */
-                if (wmb->num_arrays > 0)
-                    flush_buffer(ncid, wmb, false);
-                twmb = wmb;
-                wmb = wmb->next;
-                if (twmb == &file->buffer)
-                {
-                    twmb->ioid = -1;
-                    twmb->next = NULL;
-                }
-                else
-                {
-                    free(twmb);
-                }
-            }
-        }
-    }
-
-    /* If async is in use, send message to IO master tasks. */
-    if (ios->async)
-    {
-        int msg = PIO_MSG_SYNC;
-
-        PIO_SEND_ASYNC_MSG(ios, msg, &ierr, ncid);
-        if(ierr != PIO_NOERR)
-        {
-            LOG((1, "Error sending async msg for PIO_MSG_SYNC"));
-            return pio_err(ios, NULL, ierr, __FILE__, __LINE__);
-        }
-    }
-
-    /* Call the sync function on IO tasks. */
-    if (file->mode & PIO_WRITE)
-    {
-        if (ios->ioproc)
-        {
-            switch(file->iotype)
-            {
-#ifdef _NETCDF4
-            case PIO_IOTYPE_NETCDF4P:
-                ierr = nc_sync(file->fh);
-                break;
-            case PIO_IOTYPE_NETCDF4C:
-#endif
-#ifdef _NETCDF
-            case PIO_IOTYPE_NETCDF:
-                if (ios->io_rank == 0)
-                    ierr = nc_sync(file->fh);
-                break;
-#endif
-#ifdef _PNETCDF
-            case PIO_IOTYPE_PNETCDF:
-                flush_output_buffer(file, true, 0);
-                ierr = ncmpi_sync(file->fh);
```

Comment:
Can you remove ncmpi_sync in a separate commit (One commit that just moves all the logic to a separate function and a second commit that removes ncmpi_sync() from the new function)?